crates/matrix-sdk/src/sliding_sync/builder.rs

```diff
@@ -85,17 +84,16 @@ impl SlidingSyncBuilder {
     pub fn with_common_extensions(mut self) -> Self {
         {
             let mut cfg = self.extensions.get_or_insert_with(Default::default);
-            if cfg.to_device.is_none() {
-                cfg.to_device = Some(assign!(ToDeviceConfig::default(), { enabled: Some(true) }));
+            if cfg.to_device.is_empty() {
```

Comment:
I think this should be `cfg.to_device.enabled.is_none()`, i.e. the user has not decided whether the extension should be enabled. (effectively it is the same as `is_empty`, but there's at least theoretically the possiblity of that changing)